doc/source/data/package-ref.rst

```diff
@@ -122,6 +122,21 @@ Custom Datasource API
 .. autoclass:: ray.data.ReadTask
     :members:
 
+Partitioning API
```

Comment:
```suggestion
Datasource Partitioning API
```

To distinguish this from a Dataset Partitioning API (which may be introduced later)?